From linux-usb-devel-admin@lists.sourceforge.net Tue Jan 24 08:41:21 2006
From: Kumar Gala <galak@kernel.crashing.org>
To: linux-usb-devel@lists.sourceforge.net
Subject: USB: Fix masking bug initialization of Freescale EHCI controller
Cc: Kumar Gala <galak@gate.crashing.org>, dbrownell@users.sourceforge.net, Randy Vinson <rvinson@mvista.com>
Content-Disposition: inline
Message-Id: <200601240811.27539.david-b@pacbell.net>
Date: Tue, 24 Jan 2006 08:11:27 -0800

In setting up the of PHY we masked off too many bits, instead just
initialize PORTSC for the type of PHY we are using.

Signed-off-by: Kumar Gala <galak@kernel.crashing.org>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-fsl.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/ehci-fsl.c
+++ gregkh-2.6/drivers/usb/host/ehci-fsl.c
@@ -160,8 +160,7 @@ static void mpc83xx_setup_phy(struct ehc
 			      enum fsl_usb2_phy_modes phy_mode,
 			      unsigned int port_offset)
 {
-	u32 portsc = readl(&ehci->regs->port_status[port_offset]);
-	portsc &= ~PORT_PTS_MSK;
+	u32 portsc = 0;
 	switch (phy_mode) {
 	case FSL_USB2_PHY_ULPI:
 		portsc |= PORT_PTS_ULPI;
